backend/pages/api/ipfs/add.ts

```diff
@@ -12,46 +15,32 @@ const cors = Cors({
 type Data = {
   ipfsCid?: string;
   ipfsUrl?: string;
-  message: string;
+  message?: string;
+  detail?: string;
 };
 
-function runMiddleware(
-  req: NextApiRequest,
-  res: NextApiResponse<Data>,
-  fn: any
-) {
-  return new Promise((resolve, reject) => {
-    fn(req, res, (result: any) => {
-      if (result instanceof Error) {
-        return reject(result);
-      }
-
-      return resolve(result);
-    });
-  });
-}
-
 export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse<Data>
 ) {
-  await runMiddleware(req, res, cors);
+  await middleware(req, res, cors);
 
   if (req.method !== 'POST') {
     return res.status(405).json({ message: 'Method not allowed' });
   }
 
-  const { address: addressString, sig, metadata } = req?.body || {};
+  const { address: addressString, sig, metadata: raw } = req?.body || {};
 
-  // basic validation of message
-  if (!metadata) {
+  if (!raw) {
     return res.status(422).json({ message: 'No message' });
   }
   // validate maximum size to prevent abuse
-  const metadataJSONString = JSON.stringify(metadata);
+  const metadataJSONString = JSON.stringify(raw);
+
   if (metadataJSONString.length > MAX_MESSAGE_SIZE) {
     return res.status(422).json({ message: 'Message too long' });
   }
```

Comment:
We can get rid of this now that the JSON validation is in place.  This was just a stop-gap.